### PR TITLE
hwkagent-120 Deployment undeploy required UndeployApplicationRequest

### PR DESCRIPTION
**Pull request by jshaughn**
Provide the infrastructure to perform deployment removal via the dedicated UndeployApplicationRequest/Response messages and not as a generic operation.  Note that this is done via the server level and not directly on the deployment itself.
- Add CanonicalPath.up utility method

@pilhuhn please review

### CODE REVIEW

File: lib/hawkular/inventory/entities.rb [hunk 1]
```diff
@@ -178,39 +178,37 @@ def initialize(hash)
...
-          hash[:resource_ids].push(val)
-        when 'mt'
-          hash[:metric_type_id] = val
-        when 'rt'
-          hash[:resource_type_id] = val
-        end
+      CanonicalPath.new(path_to_h path)
+    end
+
+    def up
```
Comment (pilhuhn) [inline lib/hawkular/inventory/entities.rb:188]: What does `up` stand for?

Comment (jshaughn) [inline lib/hawkular/inventory/entities.rb:188]: It's like CanonicalPath.up() in Inventory's Java class: " Returns a path corresponding to the direct ancestor of the resource represented by this path object."  I was just adding the same API bit to this ruby class.

Comment (pilhuhn) [inline lib/hawkular/inventory/entities.rb:188]: On 5 Aug 2016, at 15:29, Jay Shaughnessy wrote:

> > +
> > -    def up
> 
> It's like CanonicalPath.up() in Inventory's Java class: " Returns a 
> path corresponding to the direct ancestor of the resource represented 
> by this path object."

Thanks. Could we name that 'parent_path' and/or add some description?

Comment (jshaughn) [inline lib/hawkular/inventory/entities.rb:188]: Sure, I'll add some doc.  I'll keep the method name the same to have parity with the java api.  I expect we can build up this ruby impl in the same way, on an as-needed basis. Commit coming...

**Comment by pilhuhn** [comment]
Thank you @jshaughn

**Pull request merged**